### PR TITLE
Tests

### PR DESCRIPTION
**Pull request by mahammad-mostafa**
- Several attribute validation rules for `User` & `Post` models
- Attributes (`name` / `title` / `posts_counter` / `comments_counter` / `likes_counter`) are required
- Attribute `title` must be smaller than or equal 250 characters
- Attributes (`posts_counter` / `comments_counter` / `likes_counter`) must be an integer and greate than or equal zero
- Unit tests for attributes validation rules and model methods

### CODE REVIEW

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  end
+  it 'posts counter should be greate than or equal zero' do
+    subject.posts_counter = -1
+    expect(subject).to_not(be_valid)
+  end
+  it 'method "recent_posts" should return last three posts associated with the user' do
+    subject.id = 1
+    Post.new(title: 'Title', author_id: subject.id, comments_counter: 0, likes_counter: 0)
+    expect(subject.recent_posts).to(eq(subject.posts))
+  end
```
Comment (MahmoudBakr23) [inline spec/models/user_spec.rb:25]: - [ ] Please divide each of your test cases with the `describe` keyword to identify each of them properly. 🙏

File: spec/models/post_spec.rb [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  end
+  it 'likes counter should be greate than or equal zero' do
+    subject.likes_counter = -1
+    expect(subject).to_not(be_valid)
+  end
+  it 'method "recent_comments" should return last five comments associated with the post' do
+    subject.id = 1
+    Comment.new(text: 'text', post_id: subject.id)
+    expect(subject.recent_comments).to(eq(subject.comments))
+  end
```
Comment (MahmoudBakr23) [inline spec/models/post_spec.rb:41]: - [ ] Please divide each of your test cases with the `describe` keyword to identify each of them properly. 🙏

**Review by MahmoudBakr23 — CHANGES_REQUESTED** [review]
Hello @mahammad-mostafa 😃, 

Great work and effort you put into this project so far.

## Highlights 🚀 

There are some excellent points that you have made perfectly. Keep up! 💪
- Well done on adding the correct validations
- Well-written validation tests
- Descriptive PR and README
- Correct Linters

However, there is always room for improvement.

> I'd like to remind you that your long hours of work are appreciated. However, the code review process is for improving your code, not criticizing it. 😄

 ## Required Changes ♻️
_Please check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

N/A

Cheers, and Happy coding!👏👨‍💻💻

Feel free to mention @MahmoudBakr23 to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Qoosim — APPROVED** [review]
#  *Status: Approved* :tada: :heavy_check_mark: 

Hi @mahammad-mostafa, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
-  Implemented required changes. :heavy_check_mark: 
-  No linter errors. :heavy_check_mark: 
-  Good README file. :heavy_check_mark: 
-  Correct Gitflow. :heavy_check_mark: 



## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**